node/node.go

```diff
@@ -2241,3 +2295,26 @@ func (w tortoiseWeakCoin) Set(lid types.LayerID, value bool) error {
 func onMainNet(conf *config.Config) bool {
 	return conf.Genesis.GenesisTime == config.MainnetConfig().Genesis.GenesisTime
 }
+
+// proposalConsumerHare is used for the hare3->hare4 migration
+// to satisfy the proposals handler dependency on hare.
+type proposalConsumerHare struct {
+	hare3          *hare3.Hare
+	h3DisableLayer types.LayerID
+	clock          *timesync.NodeClock
+	hare4          *hare4.Hare
+}
```

Comment:
I was under the impression I could get away without unit testing 1 `if` clause but I guess it won't fly :)
I was just trying to avoid a situation where now I'd have to add mocks/interfaces etc. It's no excuse for not writing the tests though...

Reply:
I wrote this comment after finding two issues with it :) Anyway, it's not a blocker as the logic is simple.